orocos_kdl/CMakeLists.txt

```diff
@@ -110,15 +108,23 @@ export(TARGETS orocos-kdl
 
 export(PACKAGE orocos_kdl)
 
-set(KDL_INCLUDE_DIRS ${KDL_INCLUDE_DIRS} ${Eigen_INCLUDE_DIR})
-
-CONFIGURE_FILE(KDLConfig.cmake.in
+# Generate CMake package configuration
+CONFIGURE_FILE(orocos_kdl-config.cmake.in
   ${PROJECT_BINARY_DIR}/orocos_kdl-config.cmake @ONLY)
-CONFIGURE_FILE(KDLConfigVersion.cmake.in
+CONFIGURE_FILE(orocos_kdl-config-version.cmake.in
   ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake @ONLY)
 
+INSTALL(FILES cmake/FindEigen3.cmake DESTINATION share/orocos_kdl/cmake)
 INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config.cmake DESTINATION share/orocos_kdl/cmake)
 INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake DESTINATION share/orocos_kdl/cmake)
 INSTALL(EXPORT OrocosKDLTargets DESTINATION share/orocos_kdl/cmake)
 
+# Generate pkg-config package configuration
+CONFIGURE_FILE(orocos_kdl.pc.in ${CMAKE_CURRENT_BINARY_DIR}/orocos-kdl.pc @ONLY)
+CONFIGURE_FILE(orocos_kdl.pc.in ${CMAKE_CURRENT_BINARY_DIR}/orocos_kdl.pc @ONLY)
+
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/orocos-kdl.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/orocos_kdl.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
```

Comment:
Should this happen twice for both lines of code?

Reply:
It used to be like this before (moved from `src/CMakeLists.txt`), that two .pc files are created and installed, one as `orocos-kdl.pc` and one as `orocos_kdl.pc`. I assume that the original package name was `orocos-kdl`, but since ROS does not support package names with dashes (`-`) the names with underscore (`_`) have been added later, in order to be compatible to rosbuild and/or catkin as build tools. For non-ROS packages using dashes as separators is actually more common.

The two `install()` commands could be merged into one, or `configure_file()` could be called only once and the same file installed twice with `install(... RENAME)`. Is that what you meant?

Reply:
I didn't noticed the dash and the underscore difference.

I would leave it the way it is.